### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by jewzaam**
* updated versions of dependencies
* fixed resource not getting closed
* fixed some race / flow control issues in Breakpoint
* fixed one API that had params swapped

### CODE REVIEW

File: migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java [hunk 1]
```diff
@@ -86,8 +88,10 @@ public void checkpoint() {
         } else {
             if(waiting) {
                 synchronized(this) {
-                    ran=true;
-                    notify();
+                    if (waiting) {
```
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java]: waiting needs to be volatile for this to work, could you change?

Comment (jewzaam) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java]: So if waiting is true before synchronized but false after, it should still notify?

Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java]: 'waiting' isn't declared as 'volatile'. It should be volatile, it doesn't change whether notify is called or not.

Comment (jewzaam) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java]: updated

**Pull request merged**